searchguard/rolesmapping.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python3
+
+import requests
+import json
+import searchguard.settings as settings
+from searchguard.exceptions import *
+from searchguard.roles import check_role_exists
+
+
+def __process_data(role, properties):
+    """Private function to process API calls for the rolemapping module"""
+    create_sg_rolemapping = requests.put('{}/rolesmapping/{}'.format(settings.SEARCHGUARD_API_URL, role),
+                                         data=json.dumps(properties),
+                                         headers=settings.HEADER,
+                                         auth=settings.SEARCHGUARD_API_AUTH)
+
+    if create_sg_rolemapping.status_code == 201 or create_sg_rolemapping.status_code == 200:
+        # Role mapping created or updated successfully
+        return
+
+    # Error when creating/updating the role mapping
+    raise RoleMappingException('Error creating/updating the mapping for role {} - msg {}'.format(
+        role, create_sg_rolemapping.text))
+
+
+def check_rolemapping_exists(role):
+    """Returns True of False depending on whether the requested role mapping exists in Search Guard"""
+    rolemapping_exists_check = requests.get('{}/rolesmapping/{}'.format(settings.SEARCHGUARD_API_URL, role),
+                                            auth=settings.SEARCHGUARD_API_AUTH)
+
+    if rolemapping_exists_check.status_code == 200:
+        # Role mapping exists in SearchGuard
+        return True
+    elif rolemapping_exists_check.status_code == 404:
+        # Role mapping does not exist in SearchGuard
+        return False
+    else:
+        # Could not fetch valid output
+        raise CheckRoleMappingExistsException('Unknown error checking whether role mapping for {} exists'.format(role))
+
+
+def view_rolemapping(role):
+    """Returns the properties for the requested role mapping if it exists"""
+    if check_rolemapping_exists(role):
```

Comment:
we're validating if the role exists before viewing it, but it means we'll send the exact same API request twice everytime the role does exist. Then I would suggest to use the same logic in both functions. Duplicating a couple of (trivial) code seems to be a good trade off to save extra HTTP calls.